### ISSUE TITLE
parfors array analysis failure

### ISSUE THREAD

**Issue by stuartarchibald** [numba/numba#3245]
Whilst updating a couple of tickets for projects relying on Numba I came across this:
https://github.com/lmcinnes/umap/issues/86
which links out to:
https://stackoverflow.com/questions/46009368/usage-of-parallel-option-in-numba-jit-decoratior-makes-function-give-wrong-resul

This code sample fails:
```python
from numba import jit
import numpy as np

@jit(nopython=True,parallel=True)
def func_2(x1,y1,x2,y2,r1,r2,n):
    x_ = np.linspace(x1,x2,n)
    y_ = np.linspace(y1,y2,n)
    z1 = np.zeros((n,n))
    for i in range(n):
        for j in range(n):
            z1[i][j] = np.sqrt((x_[i]*x_[i]+y_[j]*y_[j]))
    a = np.where((z1>(r1)) & (z1<(r2)))
    fill_factor = len(a[0])/(n*n)
    return fill_factor

x1 = 1.0
x2 = -1.0
y1 = 1.0
y2 = -1.0
r1 = 0.5
r2 = 0.75
n = 25000

func_2(x1,y1,x2,y2,r1,r2,n)
```
with:
```                                                                                                         
Traceback (most recent call last):                                                                                              
  File "sof12.py", line 24, in <module>                                                                                         
    func_2(x1,y1,x2,y2,r1,r2,n)                                                                                                 
  File "<path>/numba/dispatcher.py", line 368, in _compile_for_args                          
    raise e                                                                                                                     
  File "<path>/numba/dispatcher.py", line 325, in _compile_for_args                          
    return self.compile(tuple(argtypes))                                                                                        
  File "<path>/numba/dispatcher.py", line 657, in compile                                    
    cres = self._compiler.compile(args, return_type)                                                                            
  File "<path>/numba/dispatcher.py", line 83, in compile                                     
    pipeline_class=self.pipeline_class)                                                                                         
  File "<path>/numba/compiler.py", line 884, in compile_extra                                
    return pipeline.compile_extra(func)                                                                                         
  File "<path>/numba/compiler.py", line 369, in compile_extra                                
    return self._compile_bytecode()                                                                                             
  File "<path>/numba/compiler.py", line 815, in _compile_bytecode                            
    return self._compile_core()                                                                                                 
  File "<path>/numba/compiler.py", line 802, in _compile_core                                
    res = pm.run(self.status)                                                                                                   
  File "<path>/numba/compiler.py", line 255, in run                                          
    raise patched_exception                                                                                                     
  File "<path>/numba/compiler.py", line 246, in run                                          
    stage()                                                                                                                     
  File "<path>/numba/compiler.py", line 523, in stage_parfor_pass                            
    parfor_pass.run()                                                                                                           
  File "<path>/numba/parfor.py", line 665, in run                                            
    self.array_analysis.run(self.func_ir.blocks)                                                                                
  File "<path>/numba/array_analysis.py", line 916, in run                                    
    pre, post = self._analyze_inst(label, scope, equiv_set, inst)                                                               
  File "<path>/numba/array_analysis.py", line 949, in _analyze_inst                          
    result = self._analyze_expr(scope, equiv_set, inst.value)                                                                   
  File "<path>/numba/array_analysis.py", line 1089, in _analyze_expr                         
    return guard(fn, scope, equiv_set, expr)                                                                                    
  File "<path>/numba/ir_utils.py", line 1336, in guard                                       
    return func(*args, **kwargs)                                                                                                
  File "<path>/numba/array_analysis.py", line 1229, in _analyze_op_static_getitem            
    shape = equiv_set._get_shape(var)                                                                                           
  File "<path>/numba/array_analysis.py", line 826, in _get_shape                             
    return super(SymbolicEquivSet, self)._get_shape(obj)                                                                        
  File "<path>/numba/array_analysis.py", line 497, in _get_shape                             
    assert(vs != [])                                                                                                            
AssertionError: Failed in nopython mode pipeline (step: convert to parfors)     
```

**Comment by ehsantn** [comment]
#3211 seems to fix this issue. The problem was with this assertion in general, since it would fail if analysis is not available for some operation (`np.where` in this case).

**Comment by stuartarchibald** [comment]
@ehsantn thanks, I agree, by inspection that seems true. Please could you add a small test for this to the #3211 PR? Thanks.

**Comment by ehsantn** [comment]
Done.

**Comment by stuartarchibald** [comment]
Fixed in #3211 closing.
